### PR TITLE
SwaggerError everywhere

### PR DESCRIPTION
**Pull request by cyrylkuzmin**
This PR makes SwaggerError type as the only type of Errors in API (removes our custom Error type)
In additional:
* extend SwaggerError by Details field that contains some raw errors (like `ent: something not found`);
* move all error messages into 1 place and use these vars in handlers instead of strings;
* move some validations from repo to handlers (to generate errors properly);
* some updates in int tests: the most of error messages should be checked;

### CODE REVIEW

File: internal/handlers/common.go [hunk 1]
```diff
@@ -1,19 +1,212 @@
...
 	}
 }
 
-func buildStringPayload(msg string) *models.Error {
-	return &models.Error{
-		Data: &models.ErrorData{
-			Message: msg,
-		},
-	}
+func buildInternalErrorPayload(msg string, details string) *models.SwaggerError {
```
Comment (cyrylkuzmin) [inline internal/handlers/common.go:190]: Probably, `code` field doesn't make any sense for FE, for all our errors this code is equal to HTTP code and we can replace all this functions by 1 and use single code, for ex, 600.

File: internal/integration-tests/user/register_integration_test.go [hunk 2]
```diff
@@ -136,61 +138,103 @@ func TestIntegration_RegisterUser(t *testing.T) {
...
 
-		errExp := users.NewPostUserDefault(422)
-		errExp.Payload = &models.Error{
-			Data: nil,
+		errExp := users.NewPostUserDefault(http.StatusInternalServerError)
+		codeExp := int32(http.StatusInternalServerError)
+		errExp.Payload = &models.SwaggerError{
+			Code:    &codeExp,
+			Message: &messages.ErrCreateUser,
+			Details: "ent: validator failed for field \"User.name\": value is less than the required length",
```
Comment (cyrylkuzmin) [inline internal/integration-tests/user/register_integration_test.go:171]: Guess it should be validated on the API layer, but it's not the aim of this PR

File: internal/repositories/equipment.go [hunk 3]
```diff
@@ -637,17 +632,6 @@ func (r *equipmentRepository) UnblockEquipment(ctx context.Context, id int) erro
 	return err
 }
 
-func checkDates(start *time.Time, end *time.Time) (*time.Time, *time.Time, error) {
```
Comment (wkdjgf534) [inline internal/repositories/equipment.go:640]: Why did you remove this function?

**Review by wkdjgf534 — APPROVED** [review]

File: internal/repositories/equipment.go (hunk 3 above)
Comment (cyrylkuzmin) [reply internal/repositories/equipment.go:640]: THis validation was moved into handler to respond with proper error code and message:
```
		startDate := time.Time(s.Data.StartDate)
		endDate := time.Time(s.Data.EndDate)
		if startDate.After(endDate) {
			return equipment.NewBlockEquipmentDefault(http.StatusBadRequest).
				WithPayload(buildBadRequestErrorPayload(messages.ErrStartDateAfterEnd, ""))
		}
```
